tests/core/test_basics.py

```diff
@@ -190,3 +190,13 @@ def expected_response_type() -> Type[DummyResponse]:
 
     await asyncio.sleep(0.01)
     assert isinstance(received, DummyRequest)
+
+
+@pytest.mark.asyncio
+async def test_wait_for_can_get_cancelled(endpoint: Endpoint) -> None:
+
+    with pytest.raises(asyncio.TimeoutError):
+        await asyncio.wait_for(endpoint.wait_for(DummyRequest), 0.1)
```

Comment:
On a different note: I think `wait_for` is a poor API name. I'm usually bad with naming things but here is some inspiration:

- `wait_for_event`
- `once`
- `single`
- `single_event`

Reply:
Finding myself *mildly* resistant to adding the `_event` suffix since it feels redundant (since the argument is an event).  No strong feelings.

Reply:
I don't personally find it confusing.  Both of them map to similar concepts...

Reply:
Alright, leaving it as is then :+1: